all_deck_summary_stats.py

```diff
@@ -0,0 +1,143 @@
+import pandas as pd
+import json
+import os
+import time
+
+#import collection
+collection_directory = 'C:\\Users\\Joe\\Documents\\MTG\\'
+collection_file = '20170625 my_collection.csv'
+# import list of collection, aggregate to card name level (squashes foils/sets), then match
+collection = pd.read_csv(collection_directory + collection_file)
+collection = collection.groupby('Card')['Quantity'].sum()
+collection = pd.DataFrame(collection)
+#Add basic lands to collection
+basic_lands = pd.DataFrame([['Forest',99], ['Mountain',99], ['Island',99], ['Plains',99], ['Swamp',99]]).set_index(0)
+basic_lands = basic_lands.rename(columns={1: "Quantity"})
+collection = collection.append(basic_lands)
+
+#allocate output dataframe
+output_columns = [
+    'deck_name',
+    'archetype',
+    'url',
+    'pct_cost_owned',
+    'pct_pokedex_owned',
+    'pct_cards_owned',
+    'cost_owned',
+    'cost_total',
+    'pokedex_owned',
+    'pokedex_total',
+    'cards_owned',
+    'cards_total'
+]
+output = pd.DataFrame(columns = output_columns)
+
+#declare useful functions
+def get_df_of_cards_in_type(card_type):
+    cards_df = pd.DataFrame(list_of_card_groups[card_type])
+    cards_df['card_type'] = card_type
+    cards_df_with_good_column_names = cards_df.rename(columns={'deck-col-qty': 'how_many_the_deck_needs', 'deck-col-card': 'name', 'deck-col-price': 'price'})
+    return cards_df_with_good_column_names
+
+def get_how_many_you_have(row):
+    return collection.loc[row['name'], 'Quantity'] if row['name'] in collection.index else 0
+
+def get_how_many_you_need(row):
+    return max(row['how_many_the_deck_needs'] - row['how_many_you_have'], 0)
+
+def get_cost_to_complete_deck_requirements(row):
+    return row['how_many_you_need'] * row['unit_price']
+
+def get_value_of_cards_you_have(row):
+    return (row['how_many_the_deck_needs'] - row['how_many_you_need']) * row['unit_price']
+
+def division_by_zero_to_zero(a, b):
+    if b == 0:
+        return 0
+    else:
+        return a/b
+
+#Loop through every file
+directory = 'C:\\Users\\Joe\\Documents\\MTG\\MTG Goldfish Decks\\'
+count = 0
+number_of_decks = len(os.listdir(directory))
+
+for filename in os.listdir(directory):
+    if filename.endswith(".json"): 
+        #print(os.path.join(directory, filename))
+
+        start_time = time.time()
+        
+        json_file = directory + filename
+        
+        with open(json_file) as data_file:
+            json_data = json.load(data_file)
+        
+        deck_number = list(json_data.keys())[0]
```

Comment:
If this json parsing is specific to python 3, throw a version check at the top to prevent running in python 2.7?

something like

```
if sys.version_info[0] < 3:
    print('RUN IN PYTHON 3')
    sys.exit(1)
```